### PR TITLE
Sync 3D errorbar with 2D

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This fixes some inconsistencies of the 3D `errorbar` WRT the 2D version. Some of this is code rearrangement and cycler handling to match #17930.

Additionally, there's some inconsistency with caps for limited sides. Previously, they were drawn from a point on the line to the errorbar length with the arrow head ending on it. In 2D, the triangle marker _starts_ at the errorbar length. As a consequence of fixing this, the `arrow_length_ratio` is no longer exposed. Also, when both limits are disabled, the 2D code places both caps at zero offset, but 3D placed them at default errorbar length (this is the case for the second-last errorbar in the test image).

All that being said, the size of arrows for caps are currently calculated somewhat arbitrarily in order to match the look of the test image. Unlike 2D, the 3D quiver arrows are _not_ sized for display, but in data space. So I need to investigate that, and this will currently be marked as draft.

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 1]
```diff
@@ -2998,10 +2999,6 @@ def errorbar(self, x, y, z, zerr=None, yerr=None, xerr=None, fmt='',
             Used to avoid overlapping error bars when two series share x-axis
             values.
 
-        arrow_length_ratio : float, default: 0.4
```
Comment (tacaswell) [inline lib/mpl_toolkits/mplot3d/axes3d.py:3001]: remove from the signature as well

**Comment by tacaswell** [comment]
I am :+1: on this PR.

**Comment by QuLogic** [comment]
I'll call this done on 80%/20%, but might have a followup later.

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 2]
```diff
@@ -2996,7 +2997,7 @@ def errorbar(self, x, y, z, zerr=None, yerr=None, xerr=None, fmt='',
                  barsabove=False, errorevery=1, ecolor=None, elinewidth=None,
                  capsize=None, capthick=None, xlolims=False, xuplims=False,
                  ylolims=False, yuplims=False, zlolims=False, zuplims=False,
-                 arrow_length_ratio=.4, **kwargs):
```
Comment (tacaswell) [inline lib/mpl_toolkits/mplot3d/axes3d.py:2999]: note to other reviewers / future self: this kwarg has never been released so safe to directly remove.

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 3]
```diff
@@ -3101,34 +3098,25 @@ def errorbar(self, x, y, z, zerr=None, yerr=None, xerr=None, fmt='',
...
+        except TypeError:
+            offset = 0
 
-        base_style['label'] = '_nolegend_'
-        base_style.update(fmt_style_kwargs)
-        if 'color' not in base_style:
-            base_style['color'] = 'C0'
-        if ecolor is None:
-            ecolor = base_style['color']
+        if errorevery < 1 or int(errorevery) != errorevery:
```
Comment (tacaswell) [inline lib/mpl_toolkits/mplot3d/axes3d.py:3112]: flip order of on to avoid type errors when `errorevry` is not an int?

**Review by tacaswell — APPROVED** [review]
:+1: 

left one super minor comment.

**Comment by tacaswell** [comment]
If no one else reviews this in the next 6 or so hours I am going to merge this on a single review.

**Review by jklymak — COMMENTED** [review]
I guess I'm concerned that a lot of this is untested...  But not concerned enough to block.

**Review by anntzer — APPROVED** [review]
modulo test change discussed today.

**Comment by QuLogic** [comment]
For reference on the change in `lib/mpl_toolkits/tests/baseline_images/test_mplot3d/errorbar3d.png`, in 2D here is how the `uplim=lolim=True` case is rendered:
```
ax.errorbar(np.arange(5), np.arange(5), 1,
            lolims=[False, True, False, True, False],
            uplims=[False, False, True, True, True])
```
Figure_1

**Pull request merged**